packages/ui-components/src/components/Switch/Switch.component.tsx

```diff
@@ -0,0 +1,310 @@
+`
+
+const iconBaseStyles = `
+    jn-inline-block 
+    jn-ml-1 
+    jn-leading-1
+    jn-mt-[-.2rem]
+`
+
+const hintBaseStyles = `
+    jn-mt-0
+`
+
+type SwitchSize = "small" | "default" | "large"
+
+export interface SwitchProps extends Omit<React.ComponentPropsWithoutRef<"button">, "ref"> {
+  /**
+   * HTML name attribute for the switch button
+   */
+  name?: string
+  /**
+   * HTML id attribute for the switch button
+   */
+  id?: string
+  /**
+   * Label to display next to the Switch component
+   */
+  label?: string
+  /**
+   * Indicates if the Switch is required
+   */
+  required?: boolean
+  /**
+   * Specifies the size of the Switch component
+   */
+  size?: SwitchSize
+  /**
+   * Sets the initial checked state of the Switch component for use in uncontrolled mode
+   */
+  on?: boolean
+  /**
+   * Disabled state of the Switch component
+   */
+  disabled?: boolean
+  /**
+   * Indicates if the Switch has an error state
+   */
+  invalid?: boolean
+  /**
+   * Indicates if the Switch has been validated successfully
+   */
+  valid?: boolean
+  /**
+   * Help text to provide additional information about the Switch
+   */
+  helptext?: React.ReactNode
+  /**
+   * Text to display when the Switch has validation errors
+   */
+  errortext?: React.ReactNode
+  /**
+   * Text to display when the Switch has been validated successfully
+   */
+  successtext?: React.ReactNode
+  /**
+   * Custom class name to apply to the internal button element
+   */
+  className?: string
+  /**
+   * Callback function to handle changes to the switch state
+   */
+  // eslint-disable-next-line no-unused-vars
+  onChange?: (event: React.ChangeEvent<HTMLButtonElement>) => void
+  /**
+   * Callback function to handle click events on the switch
+   */
+  // eslint-disable-next-line no-unused-vars
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
+  /**
+   * Custom class name to apply to the wrapper element
+   * Useful for styling and positioning the outermost element of the component
+   */
+  wrapperClassName?: string
+}
+
+const isNotEmptyString = (str: unknown): str is string => {
+  return typeof str === "string" && str.trim().length > 0
+}
+
+const getSwitchSizeStyles = (size: SwitchSize): string => {
+  switch (size) {
+    case "small":
+      return "jn-w-[1.75rem] jn-h-4"
+    case "large":
+      return "jn-w-[3.125rem] jn-h-[1.6875rem]"
+    default:
+      return "jn-w-switch-default jn-h-switch-default"
+  }
+}
+
+const getHandleSizeStyles = (size: SwitchSize): string => {
+  switch (size) {
+    case "small":
+      return "jn-w-[0.75rem] jn-h-[0.75rem]"
+    case "large":
+      return "jn-w-[1.4375rem] jn-h-[1.4375rem]"
+    default:
+      return "jn-w-switch-handle-default jn-h-switch-handle-default"
+  }
+}
+
+const renderValidationIcon = (isInvalid: boolean, isValid: boolean, disabled: boolean): React.ReactNode | null => {
+  if (!isInvalid && !isValid) return null
+
+  if (isValid)
+    return (
```

Comment:
I know how to reduce a bit of the code here , coded in GitHub haha no formatting ;)
```
let icon = "dangerous"
let color = "jn-text-theme-error"

 if (isValid)
 { 
  icon = "checkCircle"
  color = "jn-text-theme-success"
 }
return (
    <Icon
      icon=icon
      color=color
      size="1.125rem"
      className={`${iconBaseStyles} ${disabled ? "jn-opacity-50" : ""}`}
    />
  )
```

Reply:
I made a small improvement but note [this bug](https://github.com/cloudoperators/juno/issues/255#issuecomment-2456557102). isValid and isInvalid can both be true at the same time for this one component 🙈😅